SixthHomework19.10.21/Counting/main.c

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+#include "circleList.h"
+#include "circleListTests.h"
+
+int giveNumberOfSurvivor(CircleList** myList, int numberOfWarriors,
+                         int killIndex)
+{
+    while (*myList != (*myList)->next)
+    {
+        int n = killIndex % numberOfWarriors + killIndex *
+                (killIndex % numberOfWarriors == 0) - 1;
+        switchToN(myList, n);
+        int errorCode = 0;
+        popNext(myList, &errorCode);
+        --numberOfWarriors;
+    }
+    return ((*myList)->value);
+}
+
+bool testGiveNumberOfSurvivor()
+{
+    CircleList* list1 = createCircleList();
+    for (int i = 0; i < 7; ++i)
+    {
+        push(&list1, i);
+    }
+    bool result = giveNumberOfSurvivor(&list1, 7, 3) == 3;
+    deleteCircleList(&list1);
+    for (int i = 0; i < 4; ++i)
+    {
+        push(&list1, i);
+    }
+    result = giveNumberOfSurvivor(&list1, 4, 5);
+    deleteCircleList(&list1);
+    return result;
+}
+
+int main() {
```

Comment:
```suggestion
int main() 
{
```
CLion так делает, но у нас стайлгайд :) CLion где-то в настройках редактора вроде переубедить можно (иначе им бы никто не пользовался, ведь у каждой компании свои стайлгайды).